### PR TITLE
refactor of resetpassword controller

### PR DESCRIPTION
**Pull request by piotr-major-rmtcfm-com**

### CODE REVIEW

File: app/controllers/Auth/ChangePasswordController.js [hunk 1]
```diff
@@ -74,14 +24,10 @@ class ResetPasswordController {
...
-        try {
-            validationResult(request).throw();
-        } catch (err) {
-            return next(err);
-        }
-
+    constructor(userRepository) {
+        this.userRepository = userRepository;
+    }
+    async invoke(request, response, next) {
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/Auth/ChangePasswordController.js]: empty line

File: app/controllers/Auth/ResetPasswordController.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+     *   @apiError (404) NotFound                The User was not found
+     *   @apiError (400) BadRequest              Email must be specified
+     */
+    constructor(userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    async invoke(request, response, next) {
+        try {
+            validationResult(request).throw();
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/Auth/ResetPasswordController.js]: please validate it via middleware

File: app/controllers/Auth/ResetPasswordController.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        } catch (err) {
+            return next(err);
+        }
+
+        const { email } = request.body;
+
+        const user = await this.userRepository.getByEmail(email);
+
+        if (!user) {
+            return response.sendStatus(HttpStatus.NOT_FOUND);
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/Auth/ResetPasswordController.js]: 204, security reason

File: app/controllers/Auth/ResetPasswordController.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        }
+        //to do - fix this
+
+        new Mail()
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/Auth/ResetPasswordController.js]: await

File: app/controllers/Auth/ResetPasswordController.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                ResetPassword({
+                    email: user.email,
+                    firstName: user.firstName,
+                    lastName: user.lastName,
+                    token: await PasswordReset.addToken(user),
+                    frontendUrl: request.get('origin')
+                })
+            )
+            .then(() => {
+                return response.sendStatus(HttpStatus.OK);
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/Auth/ResetPasswordController.js]: 204

File: app/models/User.js [hunk 6]
```diff
@@ -10,6 +10,10 @@ module.exports = (sequelize, DataTypes) => {
                 sourceKey: 'id'
             });
         }
+
+        isPasswordResetTokenExpired() {
+            return Date.now() > new Date(this.passwordResetTokenExpiresAt);
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/models/User.js]: use moment

File: app/controllers/Auth/ChangePasswordController.js [hunk 7]
```diff
@@ -74,14 +24,11 @@ class ResetPasswordController {
...
+    }
 
-    static async changePassword(request, response, next) {
-        try {
-            validationResult(request).throw();
-        } catch (err) {
-            return next(err);
-        }
-
+    async invoke(request, response, next) {
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/Auth/ChangePasswordController.js]: do you need these `next`?

File: app/controllers/Auth/ChangePasswordController.js [hunk 8]
```diff
@@ -74,14 +24,11 @@ class ResetPasswordController {
...
         const { password } = request.body;
         const token = request.params.token;
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/Auth/ChangePasswordController.js:33]: use `await` below instead of `then`

File: app/models/PasswordReset.js [hunk 9]
```diff
@@ -8,7 +8,7 @@ module.exports = (sequelize, DataTypes) => {
             PasswordReset.belongsTo(models.User, { as: 'user' });
         }
 
-        async addToken(user) {
+        static async addToken(user) {
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/models/PasswordReset.js:11]: move these methods to repo/services in next PRs

File: app/migrations/20181006095210-create-users-table.js [hunk 10]
```diff
@@ -35,6 +35,14 @@ module.exports = {
                 updatedAt: {
                     type: Sequelize.DATE,
                     allowNull: false
+                },
+                passwordResetToken: {
```
Comment (rafal-ksiazek-rmtcfm-com) [inline app/migrations/20181006095210-create-users-table.js:39]: createAt, updatedAt should be on the end

Comment (rafal-ksiazek-rmtcfm-com) [reply app/migrations/20181006095210-create-users-table.js:39]: it's standard in IT world

File: app/models/User.js [hunk 11]
```diff
@@ -1,5 +1,6 @@
 const { Model } = require('sequelize');
 const bcrypt = require('bcryptjs');
+const moment = require('moment');
```
Comment (rafal-ksiazek-rmtcfm-com) [inline app/models/User.js:3]: don't use moment, use dayjs in some PR

**Pull request merged**